File: two-sum/two-sum.py

def twoSum(nums, target):
    seen = {} # hashmap
        
    for i, num in enumerate(nums):
        complement = target - num
        if complement in seen:
            return [seen[complement], i]
        seen[num] = i
   
nums = [2, 7, 11, 19]
target = 9     
        
print(twoSum(nums,target))

File: two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

/**
 * seen[complement] -> 0 -> false
 */
var twoSum = function(nums, target) {
    const seen = {};

    for(let i=0; i< nums.length; i++) {
        const complement = target - nums[i];

        if(seen[complement] !== undefined) {
            return [seen[complement], i];
        }

        seen[nums[i]] = i;
    }
};

const nums = [2,7,11,19];
const target = 9;

console.log(twoSum(nums, target));